src/any.cpp

```diff
@@ -0,0 +1,311 @@
+      encode2(iter, std::any_cast<IntermediateAnyMapElement>(any));
+    } else {
+      assert(false); // Should not happen since we convert all map pairs that
+                     // get passed to Any to an intermediate representation.
+    }
+    break;
+  default:
+    assert(false); // In case we fail here, we forgot an implementation for a
+                   // dbus type. This is a bug then.
+    break;
+  }
+}
+
+template <typename T> void encode2(DBusMessageIter &iter, const T &val) {
+  dbus_message_iter_append_basic(&iter, get_debus_type<T>(), &val);
+}
+
+template <> void encode2(DBusMessageIter &iter, const std::string &val) {
+  const char *s = val.c_str();
+  dbus_message_iter_append_basic(&iter, DBUS_TYPE_STRING, &s);
+}
+
+template <> void encode2(DBusMessageIter &iter, const IntermediateAnyVec &val) {
+  DBusMessageIter arrayIter;
+  dbus_message_iter_open_container(&iter, DBUS_TYPE_ARRAY,
+                                   val.elementSignature.c_str(), &arrayIter);
+  for (const std::any &elem : val.vec) {
+    encodeAny(arrayIter, elem, val.elementType);
+  }
+  dbus_message_iter_close_container(&iter, &arrayIter);
+}
+
+template <>
+void encode2(DBusMessageIter &iter, const IntermediateAnyTuple &val) {
+  DBusMessageIter structIter;
+  dbus_message_iter_open_container(&iter, DBUS_TYPE_STRUCT, nullptr,
+                                   &structIter);
+  for (size_t i = 0; i < val.elements.size(); i++) {
+    encodeAny(structIter, val.elements[i], val.elementTypes[i]);
+  }
+  dbus_message_iter_close_container(&iter, &structIter);
+}
+
+template <>
+void encode2(DBusMessageIter &iter, const IntermediateAnyMapElement &val) {
+  DBusMessageIter mapElemIter;
+  dbus_message_iter_open_container(&iter, DBUS_TYPE_DICT_ENTRY, nullptr,
+                                   &mapElemIter);
+  encodeAny(mapElemIter, val.key, val.keyType);
+  encodeAny(mapElemIter, val.value, val.valueType);
+  dbus_message_iter_close_container(&iter, &mapElemIter);
+}
+
+template <> void encode2(DBusMessageIter &iter, const Any &val) {
+  if (val.containedType == DBUS_TYPE_INVALID) {
+    throw std::invalid_argument("simppl::dbus::Any can not be empty!");
+  }
+  DBusMessageIter variantIter;
+  dbus_message_iter_open_container(&iter, DBUS_TYPE_VARIANT,
+                                   val.containedTypeSignature.c_str(),
+                                   &variantIter);
+  encodeAny(variantIter, val.value_, val.containedType);
+  dbus_message_iter_close_container(&iter, &variantIter);
+}
+
+// -----------------------------------ANY-IMPLEMENTATIONS----------------------------------
+
+Any::Any(Any &&old) {
+  containedType = old.containedType;
+  containedTypeSignature = std::move(old.containedTypeSignature);
+  value_ = std::move(old.value_);
+}
+
+Any::Any(const Any &other) {
+  containedType = other.containedType;
+  containedTypeSignature = other.containedTypeSignature;
+  value_ = other.value_;
+}
```

Comment:
Copy- and move-constructor are in cpp, but copy- and move-operator are in header file.

Is there a reason to threat them differently?

Reply:
Done. Fixed.